### PR TITLE
Flex 2968 v2 device model management c88

### PR DESCRIPTION
**Pull request by MilanJansen**

### CODE REVIEW

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java [hunk 1]
```diff
@@ -57,18 +58,49 @@ public DeviceModel aDeviceModel(final Map<String, String> settings) {
...
+    /**
+     * Generic method to check if the device model is NOT created as expected in
+     * the database.
+     *
+     * @param entity
+     *            The settings.
+     * @throws Throwable
+     */
+    @Then("^the entity device model does not exists$")
+    public void theEntityDeviceModelDoesNotExists(final Map<String, String> entity) throws Throwable {
```
Comment (rlemmers) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java]: Exists => Exist

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java [hunk 2]
```diff
@@ -57,18 +58,49 @@ public DeviceModel aDeviceModel(final Map<String, String> settings) {
...
-                entity.isMetered());
+    /**
+     * Generic method to check if the device model is NOT created as expected in
+     * the database.
+     *
+     * @param entity
+     *            The settings.
+     * @throws Throwable
+     */
+    @Then("^the entity device model does not exists$")
```
Comment (rlemmers) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java]: exists => exist

**Comment by MartijnSipsCGI** [comment]
retest this please

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java [hunk 3]
```diff
@@ -57,18 +58,49 @@ public DeviceModel aDeviceModel(final Map<String, String> settings) {
...
+     */
+    @Then("^the entity device model does not exist$")
+    public void theEntityDeviceModelDoesNotExists(final Map<String, String> entity) throws Throwable {
+        final Manufacturer manufacturer = new Manufacturer();
+        final String manufacturerId = getString(entity, Keys.MANUFACTURER_ID, Defaults.DEFAULT_MANUFACTURER_ID);
+        final String modelCode = getString(entity, Keys.KEY_DEVICE_MODEL_MODELCODE,
+                Defaults.DEFAULT_DEVICE_MODEL_MODEL_CODE);
+
+        manufacturer.setManufacturerId(manufacturerId);
+        final List<DeviceModel> entityList = this.repo.findByManufacturerId(manufacturer);
```
Comment (arnerippe) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java]: Remove the manufacturerId var, insert the id directly in the set method: manufacturer.setManufacturerId(getString(entity, Keys.MANUFACTURER_ID, Defaults.DEFAULT_MANUFACTURER_ID));

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java [hunk 4]
```diff
@@ -57,18 +58,48 @@ public DeviceModel aDeviceModel(final Map<String, String> settings) {
      */
     @Then("^the entity device model exists$")
     public void theEntityDeviceModelExists(final Map<String, String> expectedEntity) throws Throwable {
+        final Manufacturer manufacturer = new Manufacturer();
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java]: I'd choose to stick code together that belongs together, so put the declaration of the Manufacturer object down to the place where it's actually needed.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java [hunk 5]
```diff
@@ -57,18 +58,48 @@ public DeviceModel aDeviceModel(final Map<String, String> settings) {
...
+     * Generic method to check if the device model is NOT created as expected in
+     * the database.
+     *
+     * @param entity
+     *            The settings.
+     * @throws Throwable
+     */
+    @Then("^the entity device model does not exist$")
+    public void theEntityDeviceModelDoesNotExists(final Map<String, String> entity) throws Throwable {
+        final Manufacturer manufacturer = new Manufacturer();
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceModelSteps.java]: I'd choose to stick code together that belongs together, so put the declaration of the Manufacturer object down to the place where it's actually needed.

**Pull request merged**